File: modules/expo-voxtral/ios/ExpoVoxtralModule.swift
import ExpoModulesCore

public class ExpoVoxtralModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoVoxtral")

    AsyncFunction("loadModel") { (modelPath: String, threads: Int, useGpu: Bool, promise: Promise) in
      DispatchQueue.global(qos: .userInitiated).async {
        let result = VoxtralBridge.shared.loadModel(path: modelPath, threads: Int32(threads), useGpu: useGpu)
        promise.resolve(result)
      }
    }

    AsyncFunction("transcribe") { (audioPath: String, promise: Promise) in
      DispatchQueue.global(qos: .userInitiated).async {
        guard VoxtralBridge.shared.isLoaded() else {
          promise.reject("ERR_NOT_LOADED", "Voxtral model is not loaded. Call loadModel() first.")
          return
        }
        if let text = VoxtralBridge.shared.transcribe(audioPath: audioPath) {
          promise.resolve(text)
        } else {
          promise.reject("ERR_TRANSCRIPTION", "Transcription failed.")
        }
      }
    }

    AsyncFunction("releaseModel") { (promise: Promise) in
      VoxtralBridge.shared.releaseModel()
      promise.resolve(nil)
    }

    Function("isModelLoaded") { () -> Bool in
      return VoxtralBridge.shared.isLoaded()
    }
  }
}

File: modules/expo-voxtral/android/src/main/java/expo/modules/voxtral/ExpoVoxtralModule.kt
package expo.modules.voxtral

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise

class ExpoVoxtralModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoVoxtral")

    AsyncFunction("loadModel") { modelPath: String, threads: Int, useGpu: Boolean, promise: Promise ->
      promise.reject("ERR_NOT_IMPLEMENTED", "Voxtral is not yet implemented on Android", null)
    }

    AsyncFunction("transcribe") { audioPath: String, promise: Promise ->
      promise.reject("ERR_NOT_IMPLEMENTED", "Voxtral is not yet implemented on Android", null)
    }

    AsyncFunction("releaseModel") { promise: Promise ->
      promise.resolve(null)
    }

    Function("isModelLoaded") {
      false
    }
  }
}
